core/src/test/scala/org/apache/spark/deploy/ExternalShuffleServiceDbSuite.scala

```diff
@@ -70,10 +71,12 @@ class ExternalShuffleServiceDbSuite extends SparkFunSuite {
     }
   }
 
+  def shuffleServiceConf: SparkConf = sparkConf.clone().set(SHUFFLE_SERVICE_PORT, 0)
```

Comment:
This change isn't exactly related; but this test was failing when I forgot a shuffle service running locally when running tests for the rest of this PR. So might as well fix this.